addons/hr_holidays/models/hr_leave.py

```diff
@@ -462,6 +470,20 @@ def _compute_request_unit_custom(self):
             if holiday.holiday_status_id or holiday.request_unit_half or holiday.request_unit_hours:
                 holiday.request_unit_custom = False
 
+    @api.depends('request_date_from', 'request_date_to')
+    def _compute_holiday_status_id(self):
+        for holiday in self:
+            leave_type = self.env['hr.leave.type'].search([])
+            date_from = fields.Datetime.to_string(holiday.request_date_from)
+            date_to = fields.Datetime.to_string(holiday.request_date_to)
+            context = dict(self.env.context).copy()
+            context.update({'default_date_from': date_from, 'default_date_to': date_to})
+            for lt in leave_type:
+                lt.with_context(context)._compute_valid()
```

Comment:
that too may prive problematic performance wise. Why not passing date_from and date_to to a dedicated method ?